### ISSUE TITLE
long, non-looping sounds don't follow the item playing them and just stay in one place

### ISSUE THREAD

**Issue by etet2** [Regalis11/Barotrauma#1026]

**Comment by etet2** [comment]
it might have something to do with the character holding it being stunned, haven't tested

**Comment by etet2** [comment]
@Regalis11 i forgot to add [dev]. add the dev lable to it

**Comment by itchyOwl** [comment]
Would be helpful to have the steps to repro to test this.

**Comment by Regalis11** [comment]
An easy way to repro is to make the submarine go very fast and activate the passive sonar. The sound will not follow the sub/sonar, but "fall behind" as the submarine moves away from the initial position of the ping. Even easier to test by changing the sonar sound to something longer.

**Comment by ekqvist** [comment]
> An easy way to repro is to make the submarine go very fast and activate the passive sonar. The sound will not follow the sub/sonar, but "fall behind" as the submarine moves away from the initial position of the ping. Even easier to test by changing the sonar sound to something longer.

I tested this by using the active sonar and by moving the sub with the debug controls outside the "level bounds" to have enough space to move the sub continuously at high speeds.

Before the fix, the sonar sound would "fall behind", as described, very noticeably when the speed was high enough. After the fix, the sound follows the sub correctly even in speeds exceeding 400 km/h. I tested it while moving to different directions and it seemed to work correctly. I would consider this Done.
